File: Practicals/07_11/Python/Palindrome.py
def main():
    str = input("Enter the String :")
    palindrome(str)

def reverse(string):
    str = ""
    for i in string:
        str = i + str
    return str

def palindrome(string):
    str = reverse(string)
    if str == string:
        print("String is Palindrome.")
    else:
        print("String is not Palindrome.")

if __name__ == "__main__":
    main()

File: Practicals/07_11/Palindrome.java
/*
 *******************************************************************************
 *  This Program is used for checking if the given string is a pelindrome or
 *  not.
 *******************************************************************************
 */

import java.util.Scanner;

class Palindrome {
    public static void main(String[] args) {
        boolean isPalindrome = true;

        Scanner str = new Scanner(System.in);

        String string = str.nextLine();
        System.out.println("String is : " + string);
        String rev = "";

        for(int i = string.length() - 1; i >= 0; i--) {
            rev = rev + string.charAt(i);
        }
        System.out.println("Reverse String is : " + rev);

        for(int i = 0; i < string.length()-1; i++) {
            if(rev.charAt(i) != string.charAt(i)) {
                isPalindrome = false;
                break;
            }
        }
        if(isPalindrome) {
            System.out.println("Palindrome...");
        }
        else {
            System.out.println("Not Palindrome...");
        }

    }
}
